### DISCUSSION TITLE
Is `joinChar` slated as an upcoming option?

### DISCUSSION THREAD

**Discussion by ArrayKnight** [react-querybuilder/react-querybuilder#787] [Q&A]
It appears that some of your array utils accept a join character with a fallback to the `defaultJoinChar` of `,`. We'd like to replace this character with something much more unique and less likely to be in natural input.

It looks like it would need to be exposed as an option in the components and hooks, added to the `toArray` utility, and replaced as the hardcoded parameter through the codebase.

Should I file an issue? Or are you already working on this? Need a hand?

For reference:
https://github.com/react-querybuilder/react-querybuilder/blob/main/packages/react-querybuilder/src/utils/arrayUtils.ts
https://github.com/react-querybuilder/react-querybuilder/blob/main/packages/react-querybuilder/src/defaults.ts#L46
https://github.com/search?q=repo%3Areact-querybuilder%2Freact-querybuilder%20joinWith&type=code
https://github.com/search?q=repo%3Areact-querybuilder%2Freact-querybuilder%20splitBy&type=code

**Comment by ArrayKnight** [comment]
Somehow I completely missed the `listsAsArrays` option which bypasses this issue: https://react-querybuilder.js.org/docs/components/querybuilder#listsasarrays

**Reply by jakeboone02** [reply]
If that works for you, then great. If you still need more control over input parsing, you could wrap `ValueEditor` and pre-process the `value` before passing the props down.

Something like this (I haven't tested this code):

```tsx
const MyValueEditor = (props: ValueEditorProps) => {
  const newValue = condition1 ? joinWith(props.value, '😀') : props.value;
  const handleOnChange = (val: any) => {
    if (condition1) {
      props.handleOnChange(splitBy(val, '😀'));
    } else {
      props.handleOnChange(val);
    }
  };

  return <ValueEditor {...props} value={newValue} handleOnChange={handleOnChange} />;
}
```

[Relevant docs](https://react-querybuilder.js.org/docs/tips/custom-with-fallback)
